src/ScyllaComics/main.ts

```diff
@@ -0,0 +1,367 @@
+            method: "GET",
+        };
+
+        const $ = await this.fetchCheerio(request);
+        return parseGenreTags($);
+    }
+
+    async getMangaDetails(mangaId: string): Promise<SourceManga> {
+        const request: Request = {
+            url: new PBURL(SCYLLA_DOMAIN)
+                .addPathComponent("manga")
+                .addPathComponent(mangaId)
+                .toString(),
+            method: "GET",
+        };
+        const $ = await this.fetchCheerio(request);
+        return parseMangaDetails($, mangaId, SCYLLA_DOMAIN);
+    }
+
+    async getChapters(sourceManga: SourceManga): Promise<Chapter[]> {
+        const request: Request = {
+            url: new PBURL(SCYLLA_DOMAIN)
+                .addPathComponent("manga")
+                .addPathComponent(sourceManga.mangaId)
+                .toString(),
+            method: "GET",
+        };
+
+        const $ = await this.fetchCheerio(request);
+        return parseChapters($, sourceManga);
+    }
+
+    async getChapterDetails(chapter: Chapter): Promise<ChapterDetails> {
+        const request: Request = {
+            url: new PBURL(SCYLLA_DOMAIN)
+                .addPathComponent("manga")
+                .addPathComponent(chapter.sourceManga.mangaId)
+                .addPathComponent(chapter.chapterId)
+                .toString(),
+            method: "GET",
+        };
+        const $ = await this.fetchCheerio(request);
+        return parseChapterDetails($, chapter);
+    }
+
+    async getSearchResults(
+        query: SearchQuery,
+        metadata: Metadata | undefined,
+    ): Promise<PagedResults<SearchResultItem>> {
+        const page: number = metadata?.page ?? 1;
+
+        const url = new PBURL("https://scyllacomics.xyz")
+            .addPathComponent("manga")
+            .setQueryItem(
+                "title",
+                encodeURIComponent(query.title?.trim() ?? ""),
+            )
+            .setQueryItem("type", "")
+            .setQueryItem("status", "")
+            .setQueryItem("page", String(page));
+
+        const getFilterValue = (id: string) =>
+            query.filters?.find((filter) => filter.id === id)?.value;
+
+        const genres =
+            (getFilterValue("genres") as Record<
+                string,
+                "included" | "excluded"
+            >) ?? {};
+
+        for (const [key, value] of Object.entries(genres)) {
+            if (value === "included") {
+                url.setQueryItem("genre[]", key);
+            }
+        }
+
+        // strict matching filter, look into getting paperback to use this with a toggle
+        const strictVal = getFilterValue("strict");
+        const isStrict =
+            typeof strictVal === "string" &&
+            (strictVal.toLowerCase() === "on" ||
+                strictVal.toLowerCase() === "true");
+        if (isStrict) {
+            url.setQueryItem("strict", "on");
+        }
```

Comment:
I would personally move the Paperback filter parsing into its own method:
```suggestion
        const getFilterValue = (id: string) =>
            query.filters?.find((filter) => filter.id === id)?.value;

        const genres =
            (getFilterValue("genres") as Record<
                string,
                "included" | "excluded"
            >) ?? {};

        for (const [key, value] of Object.entries(genres)) {
            if (value === "included") {
                url.setQueryItem("genre[]", key);
            }
        }

        // strict matching filter, look into getting paperback to use this with a toggle
        const strictVal = getFilterValue("strict");
        const isStrict =
            typeof strictVal === "string" &&
            (strictVal.toLowerCase() === "on" ||
                strictVal.toLowerCase() === "true");
        if (isStrict) {
            url.setQueryItem("strict", "on");
        }
```